libraries/lib-builtin-effects/DtmfBase.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include "PerTrackEffect.h"
+#include "ShuttleAutomation.h"
+#include <array>
+
+struct BUILTIN_EFFECTS_API DtmfSettings
+{
+   static constexpr wchar_t DefaultSequence[] = L"audacity";
+   static constexpr double DefaultDutyCycle = 55.0;
+   static constexpr double DefaultAmplitude = 0.8;
+
+   wxString dtmfSequence { DefaultSequence }; // dtmf tone string
+   size_t dtmfNTones =
+      dtmfSequence.length(); // total number of tones to generate
+   double dtmfTone {};       // duration of a single tone in ms
+   double dtmfSilence {};    // duration of silence between tones in ms
+   double dtmfDutyCycle {
+      DefaultDutyCycle
+   }; // ratio of dtmfTone/(dtmfTone+dtmfSilence)
+   double dtmfAmplitude {
+      DefaultAmplitude
+   }; // amplitude of dtmf tone sequence, restricted to (0-1)
+
+   void Recalculate(EffectSettings& settings);
+};
+
+class BUILTIN_EFFECTS_API DtmfBase :
+    public EffectWithSettings<DtmfSettings, PerTrackEffect>
+{
+public:
+   static const ComponentInterfaceSymbol Symbol;
+
+   static constexpr std::array<char, 6 * 7> kSymbols {
```

Comment:
A minor thing, what exactly `6 * 7` mean? Could we just use a simple `static constexpr std::array kSymbols {` for it to deduce type and size?

Reply:
Sorry, just seeing this, a bit late now.